packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,530 @@
+      amountSats: "0", // Temporary amount, will be adjusted later
+      locked: false,
+    }),
+  );
+
+  // Step 5: Add inputs from the original transaction
+  const originalInputTemplates = txAnalyzer.getInputTemplates();
+  const addedInputs = new Set();
+
+  if (reuseAllInputs) {
+    // Add all original inputs
+    originalInputTemplates.forEach((template) => {
+      const input = availableInputs.find(
+        (utxo) => utxo.txid === template.txid && utxo.vout === template.vout,
+      );
+      if (input) {
+        newTxTemplate.addInput(BtcTxInputTemplate.fromUTXO(input));
+        addedInputs.add(`${input.txid}:${input.vout}`);
+      }
+    });
+  } else {
+    // Add at least one input from the original transaction
+
+    // TO DO (MRIGESH)
+    // Add coin selection algorithm to add best suited input to start with ...
+    const originalInput = availableInputs.find((utxo) =>
+      originalInputTemplates.some(
+        (template) =>
+          template.txid === utxo.txid && template.vout === template.vout,
+      ),
+    );
+    if (!originalInput) {
+      throw new Error(
+        "None of the original transaction inputs found in available UTXOs.",
+      );
+    }
+    newTxTemplate.addInput(BtcTxInputTemplate.fromUTXO(originalInput));
+    addedInputs.add(`${originalInput.txid}:${originalInput.vout}`);
+  }
+
+  // Step 6: Add more inputs if necessary to meet fee requirements
+
+  // We continue adding inputs until both the fee rate is satisfied and
+  // the absolute fee meets the minimum RBF requirement
+  for (const utxo of availableInputs) {
+    if (
+      newTxTemplate.feeRateSatisfied &&
+      new BigNumber(newTxTemplate.currentFee).gte(
+        BigNumber.max(newTxTemplate.targetFeesToPay, originalTxFee),
+      )
+    ) {
+      // Stop adding inputs when both conditions are met:
+      // 1. The fee rate satisfies the target rate
+      // 2. The current fee is greater than or equal to the maximum of:
+      //    a) The target fee based on the desired rate by user
+      //    b) The minimum fee required for a valid RBF replacement >= original fees
+      break;
+    }
+    // Skip if this UTXO is already added
+    const utxoKey = `${utxo.txid}:${utxo.vout}`;
+    if (addedInputs.has(utxoKey)) {
+      continue;
+    }
+    newTxTemplate.addInput(BtcTxInputTemplate.fromUTXO(utxo));
+    addedInputs.add(utxoKey);
+  }
+
+  // Step 7: Calculate and set the cancellation output amount
+
+  const totalInputAmount = new BigNumber(newTxTemplate.getTotalInputAmount());
+  // Ensure we're using the higher of the target fee and the original fee
+  const fee = BigNumber.max(newTxTemplate.targetFeesToPay, originalTxFee);
+  // The cancel output receives all funds minus the fee
+  const cancelOutputAmount = totalInputAmount.minus(fee);
+
+  if (cancelOutputAmount.isLessThan(dustThreshold)) {
+    if (strict) {
+      throw new Error(
+        "Cancel output would be dust. Increase inputs or fee rate.",
```

Comment:
```suggestion
        "Cancel output would be dust. Provide more spendable inputs or increase fee rate.",
```

Do you maybe have to do both? I'm trying to think how you would get to this situation and how you would have to fix it.

Reply:
dust condition check removed now as we use `needsChange` method